src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -149,34 +149,34 @@ async def mypy_typecheck(
     #     `python_version`. We don't want to make the user set this up. (If they do, MyPy will use
     #     `python_version`, rather than defaulting to the executing interpreter).
     #
-    #     We only apply these optimizations if the user did not configure
-    #     `--mypy-interpreter-constraints`, and if we know that there are no Py35 or Py27
-    #     constraints. If they use Py27 or Py35, this implies that they're not using Py36+ syntax,
-    #     so it's fine to use the Py35 parser. We want the loosest constraints possible to make it
-    #     more flexible to install MyPy.
     #  * We must resolve third-party dependencies. This should use whatever the actual code's
-    #     constraints are. The constraints for the tool can be different than for the requirements.
+    #     constraints are. However, PEX will error if they are not compatible
+    #     with the interpreter being used to run MyPy. So, we should generally align the tool PEX
+    #     constraints with the requirements constraints.
+    #
+    #     However, it's possible for the requirements constraints to be different, e.g. if there is
```

Comment:
I tried to suggest a change here, but I'm not sure what "being different" means here. Different than the mypy pex? But isn't the mypy pex selected from the user's code? Do you mean different than the value of `--python-version`?

Reply:
Let me try again at rewording. I had a hard time with this because this is also likely to change again soon with partitioning.